docker-compose.yml

```diff
@@ -1208,15 +1208,13 @@ services:
       LANG: "C.UTF-8"
       BUILD_DOCS_CPP: "ON"
       BUILD_DOCS_PYTHON: "ON"
-      # GH-31506/GH-33609: Remove --disable-warnings once
-      # https://github.com/lgpage/pytest-cython/issues/24 is resolved
-      # and a new version that includes the fix is released.
-      PYTEST_ARGS: "--doctest-modules --doctest-cython --disable-warnings"
+      PYTEST_ARGS: "--doctest-modules --doctest-cython"
     volumes: *conda-volumes
     command:
       ["/arrow/ci/scripts/cpp_build.sh /arrow /build &&
         /arrow/ci/scripts/python_build.sh /arrow /build &&
         pip install -e /arrow/dev/archery[numpydoc] &&
+        pip install pytest<8 &&
```

Comment:
Can you add a comment here why this pin is needed (and maybe link to the upstream pytest-cython issue that needs to be resolved to be able to remove this)

Reply:
Sure, will do!

But I think this pin is causing CI failures. Not sure what should be changed though. I tried using double quotes (https://github.com/apache/arrow/pull/40487/commits/5ccb996cf036fa60880f9a24e6aa22ab88392c96) but it also resulted in failures: https://github.com/apache/arrow/actions/runs/8323506004/job/22773253818